### ISSUE TITLE
Add more benchmark functions

### ISSUE THREAD

**Issue by whzup** [ljvmiranda921/pyswarms#158]
Since this library is intended for researchers, I think it would be fitting to have a large range of benchmark functions at hand so one does not have to implement them himself. Good sources for such functions are:

* [Wikipedia](https://en.wikipedia.org/wiki/Test_functions_for_optimization) with a solid collection of benchmark functions and a good overview.
* [Benchmarkfcns](http://benchmarkfcns.xyz/fcns), they also have a quite big collection and even classification into different function categories (might be helpful for the documentation). They have fancy pictures and a short description for every benchmark function, as well as a MATLAB implementation.
* [Test Functions Index](http://infinity77.net/global_optimization/test_functions.html) with a ridiculous amount of benchmark functions. They even classified them by their hardness.

The functions can be implemented in the [`single_obj.py`](https://github.com/ljvmiranda921/pyswarms/blob/master/pyswarms/utils/functions/single_obj.py) file.
I think a good number to start with is 5 new benchmark functions.

Notes: Please work on the development branch. You can find a good StackOverflow question [here](https://stackoverflow.com/questions/38564473/how-to-fork-a-branch-and-not-the-repository).

**Comment by whzup** [comment]
Welcome @jayspeidell, I'm glad you're interested!
Sure go ahead! 👍 If you have any questions about the project or the GitHub process don't hesitate to comment on this issue.

**Comment by jayspeidell** [comment]
Hey @whzup! I added several new objective functions, fixed an existing one, and tested the ones that had todo abels. I'm trying to figure out the procedure and etiquette to make a pull requests. I've forked the project and made the edits on a new branch off of the development branch, and I see that there's a "PULL_REQUEST_TEMPLATE.md". Do I edit that document in-place, or save a copy somewhere, and then make the pull request? 

I also made a couple edits outside the issue, and I wanted to check in and see if they're okay. If not, none of them affect the objective function edits and I can just make a separate branch with only the changes to the single_obj.py. 

What should I do? 


Other changes: 

1. formatters/Designers

I added a colormap property to the Designer class using the same style as the rest, and type checking both types of colormap object. I did this so I could test the objective function graphs with discreet colormaps. I think it's a useful feature. 

2. plotters/plot_surface

a. I fixed a bug where creating a Designer object with default values when none exists results in an "index out of range" for the "label" property error by adding the string "z-axis" to the list. 

b. I updated Designer parameters and the line for creating a contour map to reflect the user being able to choose a colormap.

**Comment by whzup** [comment]
That's so cool @jayspeidell! 👍 You can make a PR by going into the "Pull request" section in this repository and click on "New pull request". When you do this you can choose which branches you want to compare. Be sure to choose the development branch of this repository and the branch you worked on. There you can also create a PR message, which is what the PULL_REQUEST_TEMPLATE.md is for. You can edit the PULL_REQUEST_TEMPLATE.md file as a copy and use the *source code* as PR message. GitHub can handle [Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet) (.md-files).
The most important things in the PR are the commits you want to merge. Be sure that the commit messages are well written. [Here](https://chris.beams.io/posts/git-commit/) is a link you can use as a guide. If you want to change your commits these are interesting links: [number 1](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) and [number 2](https://help.github.com/articles/changing-a-commit-message/).
[This link](https://gist.github.com/Chaser324/ce0505fbed06b947d962) gives a good guideline on what to do before you create a PR. **We work in the development, not in the master branch, though!** After you created a PR it's not the be-all and end-all. Someone's going to review the changes you did and might request changes to be done.

Regarding the other edits you did. Make sure that these are in separate commits so we have a line between the objective functions and the fixes you made in other parts of the project.

So go ahead and create a PR. I'll review it ASAP. I'm kind of busy at the moment so it might take some time 😄.

Thanks for your engagement! 💯 We really appreciate your help 😃.

**Comment by jayspeidell** [comment]
And I'm happy to help! I'm learning a lot by doing this. 

I just made the pull request and it looks like I failed a couple tests because of one issue: 

`TypeError: ("'colormap' must be (<class 'matplotlib.colors.ListedColormap'>, <class 'matplotlib.colors.LinearSegmentedColormap'>) (got [<matplotlib.colors.ListedColormap object at 0x7ff8e5409668>] that is a <class 'list'>).", Attribute(name='colormap', default=[<matplotlib.colors.ListedColormap object at 0x7ff8e5409668>], validator=<instance_of validator for type (<class 'matplotlib.colors.ListedColormap'>, <class 'matplotlib.colors.LinearSegmentedColormap'>)>, repr=True, cmp=True, hash=None, init=True, metadata=mappingproxy({}), type=None, converter=None), (<class 'matplotlib.colors.ListedColormap'>, <class 'matplotlib.colors.LinearSegmentedColormap'>), [<matplotlib.colors.ListedColormap object at 0x7ff8e5409668>])`

I have to look into this to figure it out. The 'colormap' attribute needs to validate for 'matplotlib.colors.ListedColormap' and 'matplotlib.colors.LinearSegmentedColormap' types to run properly, but it looks like the test wants it to validate for type 'list'. If it validates for type 'list' it will pass the test but fail when running. I'm kind of confused here.
